src/proc/mod.rs

```diff
@@ -97,11 +97,14 @@ impl super::TypeInner {
         }
     }
 
-    pub fn try_size(
-        &self,
-        constants: &super::Arena<super::Constant>,
-    ) -> Result<u32, crate::arena::BadHandle> {
-        Ok(match *self {
+    /// Get the size of this type.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the `constants` doesn't contain a referenced handle. This may not happen in
```

Comment:
Huh. I would have thought that it would be _less_ clear what error handling to expect from what is now the only public way to get the size of a type. 🤔

In general, I've found it more beneficial to document the conditions under which a public API could panic. You're saying that you don't think that that outweighs the maintenance cost, in this case?

Reply:
(OT, but still most relevant to this convo out of this whole PR: remove the `the` article with c533142a.)

Reply:
I've removed the `Panics` section with 65251d65.

Reply:
> In general, I've found it more beneficial to document the conditions under which a public API could panic. You're saying that you don't think that that outweighs the maintenance cost, in this case?

That's a fair question, and it's a point that comes up in many contexts, so let me explain my thinking:

The information content of a message is the -log₂ of the probability of its being received: for example, a message that is received with probability 0.25 conveys two bits of information. But this means that a message that is almost always received has a probability approaching 1, and thus conveys almost zero bits. Hearing "good morning" from someone who always says that is nice, but it doesn't tell you much.

Almost every public Naga API that operates on a module or a part of a module is going to access arenas, and thus will panic in the presence of invalid handles. If we were to place these panic notices correctly, almost every public API in Naga would have a "Panics" section saying the same thing. That would be a message with a probability approaching one, and thus one that conveys little information, and thus one that might be best omitted, since the reader has a limited attention span.

However, if we document this general approach in higher-level documentation, like the lib.rs comments on `Arena`s, that's being read by users who are getting an overview of Naga and are new to its API. Such readers are probably not ready to make many assumptions about Naga's behavior, so explaining that validation checks for bad handles, but subsequent stages assume these checks have been done and will panic (but not exhibit UB) if presented with bad handles, is not at all expected, and would convey many bits. So putting the message in a place where it has the most effect on the readers' understanding improves its information content.

The Rust standard library documentation for slice methods has a "Panics" section on every method that takes an index. For example, `split_at(&self, mid: usize)` says, "Panics if mid > len." I *think* this situation is different from Naga's, because there are various ways to respond to an index that is out of bounds. You could truncate (like the `windows` method when `size` is too large), it could be UB (like `split_unchecked`), etc. But in Naga, a bad handle is always meaningless. So there's really no other possible response.

In any case, if we're going to mention the panics, then I want to do it systematically - and that's not for this PR.